### PR TITLE
Replace `npm` with `yarn`

### LINKED ISSUE #5878: Replace `npm` with `yarn` in `CONTRIBUTING.rst`
https://github.com/mlflow/mlflow/blob/master/CONTRIBUTING.rst#javascript-and-ui needs to be updated because we no longer use `npm` but currently use `yarn`.

```
> cat CONTRIBUTING.rst | grep npm
The MLflow UI is written in JavaScript. ``npm`` is required to run the Javascript dev server and the tracking UI.
You can verify that ``npm`` is on the PATH by running ``npm -v``, and
`install npm <https://www.npmjs.com/get-npm>`_ if needed.
   npm install
If modifying dependencies in ``mlflow/server/js/package.json``, run ``npm update`` within
   npm start
   npm run build
```

### PR DESCRIPTION
**Pull request by kugland**
## Related Issues/PRs

Close #5878 

## What changes are proposed in this pull request?
Replaced `npm` with `yarn` in CONTRIBUTING.rst and in the issue template.

## How is this patch tested?
No need to test, just a change in a text file.

## Does this PR change the documentation?
- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report_template.yaml [hunk 1]
```diff
@@ -31,7 +31,7 @@ body:
         - **MLflow installed from (source or binary)**:
         - **MLflow version (run ``mlflow --version``)**:
         - **Python version**:
-        - **npm version, if running the dev UI**:
+        - **yarn version, if running the dev UI**:
```
Comment (harupy) [inline .github/ISSUE_TEMPLATE/bug_report_template.yaml:34]: Good catch!

**Comment by rafaelvp-db** [comment]
nice work @kugland. in my case I also needed to do a quick workaround by running ```touch yarn.lock``` so perhaps we want to add that to the instructions as well.

**Comment by harupy** [comment]
@rafaelvp-db

> nice work @kugland. in my case I also needed to do a quick workaround by running `touch yarn.lock` so perhaps we want to add that to the instructions as well.

Did you run `yarn install` in `mlflow/server/js`?

**Comment by harupy** [comment]
@kugland Can you fix the DCO check failure? https://github.com/mlflow/mlflow/pull/5879/checks?check_run_id=6451023747

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -553,7 +553,7 @@ Generate JS files in ``mlflow/server/js/build``:
 .. code-block:: bash
 
    cd mlflow/server/js
-   npm run build
+   yarn run build
```
Comment (harupy) [inline CONTRIBUTING.rst:556]:
```suggestion
   yarn build
```

can we remove `run`?

**Comment by kugland** [comment]
> @rafaelvp-db
> 
> > nice work @kugland. in my case I also needed to do a quick workaround by running `touch yarn.lock` so perhaps we want to add that to the instructions as well.
> 
> Did you run `yarn install` in `mlflow/server/js`? Can you share more info about the issue you encountered?

I didn’t run into any hard error installing the dependencies, but I’ve got quite a few warnings (using yarn 3.1.1):

```➤ YN0000: ┌ Resolution step
➤ YN0002: │ @craco/craco@npm:7.0.0-alpha.0 [5ace8] doesn't provide @types/node (p80a72), requested by cosmiconfig-typescript-loader
➤ YN0002: │ @craco/craco@npm:7.0.0-alpha.0 [5ace8] doesn't provide typescript (pa2860), requested by cosmiconfig-typescript-loader
➤ YN0002: │ @databricks/design-system@npm:1.0.0 [5ace8] doesn't provide rollup (pd4b60), requested by @rollup/plugin-node-resolve
➤ YN0002: │ @mdx-js/loader@npm:1.6.22 doesn't provide react (p12550), requested by @mdx-js/react
➤ YN0002: │ @storybook/preset-create-react-app@npm:4.1.0 [5ace8] doesn't provide react-refresh (p7423a), requested by @pmmmwh/react-refresh-webpack-plugin
➤ YN0002: │ @storybook/preset-create-react-app@npm:4.1.0 [5ace8] doesn't provide typescript (pe30f7), requested by @storybook/react-docgen-typescript-plugin
➤ YN0002: │ @storybook/preset-create-react-app@npm:4.1.0 [5ace8] doesn't provide webpack (p07b0b), requested by @pmmmwh/react-refresh-webpack-plugin
➤ YN0002: │ @storybook/preset-create-react-app@npm:4.1.0 [5ace8] doesn't provide webpack (p08e01), requested by @storybook/react-docgen-typescript-plugin
➤ YN0002: │ eslint-config-react-app@npm:7.0.0 [1920f] doesn't provide @babel/plugin-syntax-flow (peddab), requested by eslint-plugin-flowtype
➤ YN0002: │ eslint-config-react-app@npm:7.0.0 [1920f] doesn't provide @babel/plugin-transform-react-jsx (p350d1), requested by eslint-plugin-flowtype
➤ YN0002: │ js@workspace:. doesn't provide @babel/core (pfff3d), requested by @babel/eslint-parser
➤ YN0002: │ js@workspace:. doesn't provide @babel/core (pa2339), requested by @storybook/addon-essentials
➤ YN0002: │ js@workspace:. doesn't provide @babel/core (p9f319), requested by @storybook/preset-create-react-app
➤ YN0002: │ js@workspace:. doesn't provide @babel/core (p1581f), requested by babel-jest
➤ YN0002: │ js@workspace:. doesn't provide @typescript-eslint/parser (pdeb9b), requested by @typescript-eslint/eslint-plugin
➤ YN0002: │ js@workspace:. doesn't provide @typescript-eslint/parser (p66501), requested by eslint-config-react-app
➤ YN0002: │ js@workspace:. doesn't provide babel-eslint (p42528), requested by eslint-config-react-app
➤ YN0002: │ js@workspace:. doesn't provide babel-loader (p13fe1), requested by @storybook/addon-essentials
➤ YN0060: │ js@workspace:. provides eslint (p72b85) with version 7.23.0, which doesn't satisfy what eslint-config-airbnb-base requests
➤ YN0060: │ js@workspace:. provides eslint (pdecbb) with version 7.23.0, which doesn't satisfy what eslint-loader requests
➤ YN0060: │ js@workspace:. provides react (p3fc08) with version 17.0.2, which doesn't satisfy what antd and some of its descendants request
➤ YN0060: │ js@workspace:. provides react (p39ee8) with version 17.0.2, which doesn't satisfy what enzyme-adapter-react-16.3 and some of its descendants request
➤ YN0060: │ js@workspace:. provides react (p330a3) with version 17.0.2, which doesn't satisfy what react-mde requests
➤ YN0060: │ js@workspace:. provides react (p8406f) with version 17.0.2, which doesn't satisfy what react-pdf requests
➤ YN0060: │ js@workspace:. provides react (p84060) with version 17.0.2, which doesn't satisfy what react-treebeard and some of its descendants request
➤ YN0060: │ js@workspace:. provides react (p841de) with version 17.0.2, which doesn't satisfy what react-virtualized requests
➤ YN0060: │ js@workspace:. provides react-dom (p972a9) with version 17.0.2, which doesn't satisfy what antd and some of its descendants request
➤ YN0060: │ js@workspace:. provides react-dom (pb6e35) with version 17.0.2, which doesn't satisfy what enzyme-adapter-react-16.3 requests
➤ YN0060: │ js@workspace:. provides react-dom (p50777) with version 17.0.2, which doesn't satisfy what react-mde requests
➤ YN0060: │ js@workspace:. provides react-dom (pa0903) with version 17.0.2, which doesn't satisfy what react-pdf requests
➤ YN0060: │ js@workspace:. provides react-dom (pc0c37) with version 17.0.2, which doesn't satisfy what react-treebeard and some of its descendants request
➤ YN0060: │ js@workspace:. provides react-dom (p924ec) with version 17.0.2, which doesn't satisfy what react-virtualized requests
➤ YN0060: │ js@workspace:. provides redux (pcb2c5) with version 4.1.2, which doesn't satisfy what redux-promise-middleware requests
➤ YN0060: │ js@workspace:. provides webpack (pdefad) with version 5.72.0, which doesn't satisfy what eslint-loader requests
➤ YN0002: │ react-dev-utils@npm:11.0.4 doesn't provide typescript (p79ddf), requested by fork-ts-checker-webpack-plugin
➤ YN0002: │ react-dev-utils@npm:11.0.4 doesn't provide webpack (p2af19), requested by fork-ts-checker-webpack-plugin
➤ YN0002: │ react-dev-utils@npm:12.0.0 doesn't provide typescript (p08c91), requested by fork-ts-checker-webpack-plugin
➤ YN0002: │ react-dev-utils@npm:12.0.0 doesn't provide webpack (pf80ce), requested by fork-ts-checker-webpack-plugin
➤ YN0002: │ react-pdf@npm:4.1.0 [5ace8] doesn't provide webpack (p0e051), requested by pdfjs-dist
➤ YN0002: │ react-scripts@npm:5.0.0 [5ace8] doesn't provide autoprefixer (p1a9b7), requested by tailwindcss
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed in 0s 696ms
➤ YN0000: ┌ Fetch step
➤ YN0013: │ yargs@npm:3.10.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yn@npm:3.1.1 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ yocto-queue@npm:0.1.0 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ zero-crossings@npm:1.0.1 can't be found in the cache and will be fetched from the remote registry
➤ YN0013: │ zwitch@npm:1.0.5 can't be found in the cache and will be fetched from the remote registry
➤ YN0000: └ Completed in 1m 51s
➤ YN0000: ┌ Link step
➤ YN0076: │ fsevents@patch:fsevents@npm%3A2.3.2#~builtin<compat/fsevents>::version=2.3.2&hash=18f3a7 The linux-x64 architecture is incompatible with this module, link skipped.
➤ YN0076: │ fsevents@patch:fsevents@npm%3A2.1.3#~builtin<compat/fsevents>::version=2.1.3&hash=18f3a7 The linux-x64 architecture is incompatible with this module, link skipped.
➤ YN0076: │ fsevents@patch:fsevents@npm%3A1.2.9#~builtin<compat/fsevents>::version=1.2.9&hash=18f3a7 The linux-x64 architecture is incompatible with this module, link skipped.
➤ YN0007: │ core-js-pure@npm:3.19.3 must be built because it never has been before or the last one failed
➤ YN0007: │ core-js@npm:3.21.1 must be built because it never has been before or the last one failed
➤ YN0007: │ core-js@npm:2.6.12 must be built because it never has been before or the last one failed
➤ YN0007: │ core-js-pure@npm:3.20.2 must be built because it never has been before or the last one failed
➤ YN0007: │ core-js-pure@npm:3.15.1 must be built because it never has been before or the last one failed
➤ YN0000: └ Completed in 1m 33s
➤ YN0000: Done with warnings in 3m 25s
```

**Comment by harupy** [comment]
I'm getting the exact same warnings. We can ignore them.

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**